src/smart-components/workflow/add-stages/set-stages.js

```diff
@@ -22,54 +26,74 @@ const SetStages = (formData, onHandleChange, options) => {
     let values = stageValues;
     values[index] = value;
     setStageValues(values);
-    onHandleChange({ wfGroups: values });
+    handleChange({ wfGroups: values });
+  };
+
+  const addStage = () => {
+    setStageValues([ ...stageValues, undefined ]);
+    setStageIndex(stageIndex + 1);
+  };
+
+  const removeStage = (idx) => {
+    const i = parseInt(idx.target.id);
+    const values = [ ...stageValues.filter((_, j) => i !== j) ];
+    setStageValues(values);
+    setStageIndex(stageIndex - 1);
+    handleChange({ wfGroups: values });
   };
 
   const createStageInput = (idx) => {
     return (
       <StackItem>
-        <FormSelect
-          label={ `${idx + 1} Stage` }
-          aria-label= { `${idx + 1} Stage` }
-          onToggle={ onToggle }
-          key = { `stage-${idx + 1}` }
-          onChange={ (e) => onStageChange(e, idx) }
-          value={ stageValues[idx] }
-          isExpanded={ isExpanded }
-          ariaLabelledBy={ `Stage-${idx}` }
+        <FormGroup
+          label={ `Stage ${idx + 1}` }
+          fieldId={ `${idx + 1}_stage_label` }
         >
-          { options.map((option) => (
-            <FormSelectOption
-              isDisabled={ option.disabled }
-              key={ option.value || option.label }
-              label={ option.label.toString() }
-              value={ option.value }
-              isPlaceholder={ option.isPlaceholder }
-            />
-          )) }
-        </FormSelect>
+          <Grid gutter="md">
+            <GridItem span={ 8 }>
+              <FormSelect
+                label={ `${idx + 1} Stage` }
+                aria-label={ `${idx + 1} Stage` }
+                onToggle={ onToggle }
+                key={ `stage-${idx + 1}` }
+                onChange={ (e) => onStageChange(e, idx) }
+                value={ stageValues[idx] }
+                isexpanded={ isExpanded }
+                aria-labelledby={ `Stage-${idx}` }
+              >
+                { options.map((option) => (
+                  <FormSelectOption
+                    isdisabled={ option.disabled }
+                    key={ option.value || option.label }
+                    label={ option.label.toString() }
+                    value={ option.value }
+                    isplaceholder={ option.isPlaceholder }
+                  />
+                )) }
+              </FormSelect>
+            </GridItem>
+            <GridItem span={ 1 }>
```

Comment:
I noticed small vertical misalignment on the remove icon. If you add `display: flex` to the GridItem style i will alight the icon to the grid item baseline.

![screenshot-ci foo redhat com-1337-2019 08 09-08-39-05](https://user-images.githubusercontent.com/22619452/62759286-28d36180-ba81-11e9-966a-1c4c0c7935dd.png)

Reply:
Thank you Martin - updated.